src/lib/FlightTasks/tasks/ManualAltitude/FlightTaskManualAltitude.cpp

```diff
@@ -84,6 +96,25 @@ void FlightTaskManualAltitude::_scaleSticks()
 	// scale horizontal velocity with expo curve stick input
 	const float vel_max_z = (_sticks(2) > 0.0f) ? _constraints.speed_down : _constraints.speed_up;
 	_velocity_setpoint(2) = vel_max_z * _sticks_expo(2);
+
+	// if there is a valid distance to bottom or distance to home, then
+	// adjust speed downwards gradually within the limits MPC_LAND_ALT1 and MPC_LAND_ALT2.
+	if (_sticks(2) > 0.0f) { //user demands speed downwards
+
+		float dist_to_ground = NAN;
+
+		if (PX4_ISFINITE(_dist_to_bottom)) {
+			dist_to_ground = _dist_to_bottom;
+
+		} else if (PX4_ISFINITE(_sub_home_position->get().valid_alt)) {
+			dist_to_ground = -_position(2) + _sub_home_position->get().z;
+		}
+
+		if (PX4_ISFINITE(dist_to_ground)) {
+			_velocity_setpoint(2) = math::gradual(dist_to_ground,  MPC_LAND_ALT2.get(),
```

Comment:
This will not generate a linear setpoint change as the slope is defined by the actual - changing - velocity setpoint: `_velocity_setpoint = f(_velocity_setpoint, dist_to_ground)`. Is that the required behavior?